### PR TITLE
mortality observer

### PR DESCRIPTION
**Pull request by hussain-jafari**
### Description
- *Category*: feature, test
- *JIRA issue*: [MIC-3950](https://jira.ihme.washington.edu/browse/MIC-3950)

### Changes and notes
Update mortality observer for new results management system and add tests. Exit year stratification registration and its use in tests had to be commented out to get builds passing. This is a known bug which should already have a ticket but I'll make one if it doesn't exist. 

### Testing
Ran simulation for a few time steps and tests passed.

### CODE REVIEW

File: src/vivarium_public_health/metrics/mortality.py [hunk 1]
```diff
@@ -71,99 +67,53 @@ def name(self):
...
+                    "exit_time",
+                    "years_of_life_lost",
+                ],
+                additional_stratifications=self.config.include,
+                excluded_stratifications=self.config.exclude,
+                when="collect_metrics",
+            )
+
+    def count_cause_specific_deaths(self, x: pd.DataFrame) -> float:
+        died_of_cause = x["exit_time"] == self.clock() + self.step_size()
```
Comment (rmudambi) [inline src/vivarium_public_health/metrics/mortality.py:114]: Why not just `died_of_cause = x["exit_time"] > self.clock()`? Then you don't need to store `self.step_size`. The same applies below in YLLs

File: tests/metrics/test_mortality_observer.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        }
+    }
+
+
+def disease_with_excess_mortality(base_config, disease_name) -> DiseaseModel:
+    year_start = base_config.time.start.year
+    year_end = base_config.time.end.year
+    healthy = SusceptibleState(disease_name)
+    disease_get_data_funcs = {
+        "disability_weight": lambda _, __: build_table(0.0, year_start - 1, year_end),
```
Comment (rmudambi) [inline tests/metrics/test_mortality_observer.py:30]: Nit: you can do `lambda *_:` instead

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_public_health/metrics/stratification.py [hunk 3]
```diff
@@ -60,13 +60,13 @@ def register_stratifications(self, builder: Builder) -> None:
...
-            is_vectorized=True,
-            requires_columns=["exit_time"],
-        )
+        # builder.results.register_stratification(
+        #     "exit_year",
+        #     [str(year) for year in range(self.start_year, self.end_year + 1)] + ["nan"],
+        #     self.map_year,
+        #     is_vectorized=True,
+        #     requires_columns=["exit_time"],
+        # )
```
Comment (mattkappel) [inline src/vivarium_public_health/metrics/stratification.py:69]: ?

File: tests/metrics/test_stratification.py [hunk 4]
```diff
@@ -138,17 +138,17 @@ def test_results_stratifier_register_stratifications(mocker):
...
-        is_vectorized=True,
-        requires_columns=["exit_time"],
-    )
+    # builder.results.register_stratification.assert_any_call(
+    #     "exit_year",
+    #     years_list + ["nan"],
+    #     rs.map_year,
+    #     is_vectorized=True,
+    #     requires_columns=["exit_time"],
+    # )
```
Comment (mattkappel) [inline tests/metrics/test_stratification.py:147]: Also, ?

**Review by mattkappel — COMMENTED** [review]
Approved!

**Pull request merged**